File: Clothing.java
public class Clothing extends Product {
    private String size, material, gender;

    public Clothing()
    {
        this.size = "";
        this.material = "";
        this.gender = "";
    }

    public Clothing(String size, String material, String gender)
    {
        this.size = size;
        this.material = material;
        this.gender = gender;
    }

    public String getSize() {
        return this.size;
    }

    public void setSize(String size) {
        this.size = size;
    }

    public String getMaterial() {
        return this.material;
    }

    public void setMaterial(String material) {
        this.material = material;
    }

    public String getGender() {
        return this.gender;
    }

    public void setGender(String gender) {
        this.gender = gender;
    }
}

File: prak php/Clothing.php
<?php
require('Product.php');

class Clothing extends Product {
    private $size, $material, $gender;

    // public function __construct() {
    //     parent::__construct(); // Memanggil konstruktor kelas induk
    //     $this->size = "";
    //     $this->material = "";
    //     $this->gender = "";
    // }

    public function __construct($size="", $material="", $gender="") {
        parent::__construct(); // Memanggil konstruktor kelas induk
        $this->size = $size;
        $this->material = $material;
        $this->gender = $gender;
    }

    public function getSize() {
        return $this->size;
    }

    public function setSize($size) {
        $this->size = $size;
    }

    public function getMaterial() {
        return $this->material;
    }

    public function setMaterial($material) {
        $this->material = $material;
    }

    public function getGender() {
        return $this->gender;
    }

    public function setGender($gender) {
        $this->gender = $gender;
    }
}

?>